specification/appplatform/resource-manager/Microsoft.AppPlatform/preview/2024-05-01-preview/appplatform.json

```diff
@@ -15168,6 +15868,296 @@
+          "type": "integer"
+        }
+      }
+    },
+    "JobResourceCollection": {
+      "description": "List of Azure Spring Apps Jobs and a possible link for next set.",
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "Collection of Job resources.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/JobResource"
+          }
+        },
+        "nextLink": {
+          "description": "Link to next page of resources.",
+          "type": "string"
+        }
+      }
+    },
+    "JobExecutionTemplate": {
+      "description": "Job's execution template, containing configuration for an execution",
+      "type": "object",
+      "properties": {
+        "environmentVariables": {
+          "description": "Environment variables of Job execution",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/EnvVar"
+          },
+          "x-ms-identifiers": [
+            "name"
+          ]
+        },
+        "args": {
+          "description": "Arguments for the Job execution.",
+          "uniqueItems": false,
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "resourceRequests": {
+          "$ref": "#/definitions/JobResourceRequests",
+          "description": "The requested resource quantity for required CPU and Memory."
+        }
+      }
+    },
+    "JobExecution": {
+      "description": "Azure Spring Apps Job execution.",
+      "type": "object",
+      "readOnly": true,
+      "properties": {
+        "name": {
+          "description": "Job execution Name.",
+          "type": "string"
+        },
+        "status": {
+          "description": "Current state of the job execution",
+          "enum": [
+            "Running",
+            "Pending",
+            "Canceled",
+            "Failed",
+            "Completed"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "JobExecutionRunningState",
+            "modelAsString": true
+          }
+        },
+        "startTime": {
+          "description": "Job execution start time.",
+          "format": "date-time",
+          "type": "string"
+        },
+        "endTime": {
+          "description": "Job execution end time.",
+          "format": "date-time",
+          "type": "string"
+        },
+        "template": {
+          "$ref": "#/definitions/JobExecutionTemplate",
+          "description": "The template which is applied for the execution of the Job."
+        },
+        "jobSnapshot": {
+          "$ref": "#/definitions/JobResourceProperties",
+          "description": "The snapshot of job configuration"
+        }
+      }
+    },
+    "JobExecutionCollection": {
+      "description": "Azure Spring App Job executions collection.",
+      "type": "object",
+      "properties": {
+        "value": {
```

Comment:
please add `x-ms-identifiers` for array type:
https://github.com/Azure/azure-openapi-validator/blob/main/docs/xms-identifier-validation.md

Reply:
added